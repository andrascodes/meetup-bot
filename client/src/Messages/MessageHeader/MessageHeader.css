.MessageHeader {
  padding: 2em;
  border-bottom: 0.1em solid rgba(0, 0, 0, .10);
  margin-bottom: 1em;
  color: rgba(0, 0, 0, 1);
  font-size: 14px;
}