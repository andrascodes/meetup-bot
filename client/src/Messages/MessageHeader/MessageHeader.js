import React from 'react'
import { Link } from 'react-router-dom'

import { formatDate } from '../../lib/dateManipulation'

import './MessageHeader.css'

export const MessageHeader = props => (
  <div className="MessageHeader">
    <div className="container-fluid">
      <div className="row">
        <div className="col-lg-4 col-md-4 col-sm-4 text-left">
          <Link to={{
            pathname: '/conversations',
            state: props.filterState
          }}>Back</Link>
        </div>
        <div className="col-lg-4 col-md-4 col-sm-4 text-center">{props.participant}</div>
        <div className="col-lg-4 col-md-4 col-sm-4 text-right">{formatDate(props.startDate)}</div>
      </div>
    </div>
  </div>
)