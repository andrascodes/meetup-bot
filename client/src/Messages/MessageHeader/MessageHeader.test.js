import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

import { MessageHeader } from './MessageHeader'

describe('MessageHeader', () => {

  test('renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <MessageHeader 
        />
      </MemoryRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  test('renders correctly with props', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <MessageHeader 
          filterState={undefined}
          participant="123"
          startDate="2017-04-28 12:05"
          label="label1 label2 label3"
        />
      </MemoryRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})