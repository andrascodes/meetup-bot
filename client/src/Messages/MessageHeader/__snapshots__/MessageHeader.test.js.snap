exports[`MessageHeader renders correctly 1`] = `
<div>
  <div
    className="MessageHeader">
    <div
      className="container-fluid">
      <div
        className="row">
        <div
          className="col-lg-4 col-md-4 col-sm-4 text-left">
          <a
            href="/conversations"
            onClick={[Function]}>
            Back
          </a>
        </div>
        <div
          className="col-lg-4 col-md-4 col-sm-4 text-center header-item">
          <div />
        </div>
        <div
          className="col-lg-4 col-md-4 col-sm-4 text-right header-item" />
      </div>
    </div>
  </div>
  <div
    className="Label text-center">
    Label: 
  </div>
</div>
`;

exports[`MessageHeader renders correctly with props 1`] = `
<div>
  <div
    className="MessageHeader">
    <div
      className="container-fluid">
      <div
        className="row">
        <div
          className="col-lg-4 col-md-4 col-sm-4 text-left">
          <a
            href="/conversations"
            onClick={[Function]}>
            Back
          </a>
        </div>
        <div
          className="col-lg-4 col-md-4 col-sm-4 text-center header-item">
          <div>
            123
          </div>
        </div>
        <div
          className="col-lg-4 col-md-4 col-sm-4 text-right header-item">
          April 28th 2017 12:05
        </div>
      </div>
    </div>
  </div>
  <div
    className="Label text-center">
    Label: 
    label1 label2 label3
  </div>
</div>
`;
